Tests/ParseTests.cs

```diff
@@ -1315,11 +1315,57 @@ override void visit(ClassNode node)
 
 			var mod = DParser.ParseString (code);
 			Assert.That (mod.ParseErrors.Count, Is.EqualTo (0));
-		}
-
-
+		}
+
+		[Test]
+		public void EnumAttributes()
+		{
+			var m = DParser.ParseString(@"enum E { @disable Dis, deprecated Dep, @1234 uda }");
+
+			Assert.AreEqual(0, m.ParseErrors.Count);
+
+			var E = m["E"].First() as DNode;
+			Assert.That(E is DEnum);
+
+			var Dis = (E as DEnum)["Dis"].First() as DNode;
+			Assert.That(Dis is DEnumValue);
+			var attr = Dis.Attributes;
+			Assert.That (attr.Count == 1);
+			Assert.That(attr[0] is BuiltInAtAttribute);
+			Assert.That((attr[0] as BuiltInAtAttribute).Kind == BuiltInAtAttribute.BuiltInAttributes.Disable);
+
+			var Dep = (E as DEnum)["Dep"].First() as DNode;
+			Assert.That(Dep is DEnumValue);
+			attr = Dep.Attributes;
+			Assert.That(attr.Count == 1);
+			Assert.That(attr[0] is DeprecatedAttribute);
+
+			var uda = (E as DEnum)["uda"].First() as DNode;
+			Assert.That(uda is DEnumValue);
+			attr = uda.Attributes;
+			Assert.That(attr.Count == 1);
+			Assert.That(attr[0] is AtAttribute);
+		}
+
+		[Test]
+		public void ParameterUDA()
+		{
+			var m = DParser.ParseString(@"void example(@(22) string param) {}");

```

Comment:
Das funktionierte offensichtlich schon, aber das habe ich erst gemerkt, nachdem ich den Test geschrieben hatte.

Reply:
:+1: :grinning:

Reply:
Bin ich es eigentlich, der immer CRLF in den Source hinzufügt? Ich bekomme ab und zu Meldungen von VS wegen "mixed line endings", zum Teil aber auch auf Dateien, die ich meine, noch nie angefasst zu haben.

Reply:
Eigentlich ändert der Parser nichts an Dateien - gibt es nicht pro Datei Lineending-Einstellungen, die man ändern könnte?
-- Warum sollte man andererseits unter windows nicht crlf verwenden? :D
Es kann sein, dass vielleicht irgendein Formatter Dinge anfasst?